### ISSUE TITLE
Feature request: fbinfer static analysis for C, Objective-C and Java

### ISSUE THREAD

**Issue by thyrgle** [scrooloose/syntastic#1476]
[Infer](http://fbinfer.com/) is a relatively new static analysis tool for analyzing C, Objective-C, and Java programs. I have tried implementing a C plug-in for infer (as seen by [this stackoverflow question](http://stackoverflow.com/questions/31415482/using-syntastic-and-infer)), but I am relatively new to vimscript, and don't exactly know what I am doing. Is anyone willing to add this syntastic or help me out?

Thanks.

**Comment by lcd047** [comment]
I'm afraid `Infer` doesn't fit very well with the way syntastic does things.  First, it's a global analysis tool.  When `Infer` points to an error you typically have to go through several files to understand what's going on, and then the real problem is somewhere else altogether.  Syntastic is good at dealing with _syntax_ errors, which typically can be fixed "in place".  `Infer` has the `inferTraceBugs` to help you, and there's no way to interface that with syntastic.

Then, while `Infer` _can_ analyse single files, it's typical use is to capture the results of full builds and analyse the results.  That, again, doesn't mesh well with syntastic.

There are also a number of other annoyances, like having to deal with an output directory ~3 times larger than the project itself, having to parse column numbers from messages rather than from a dedicated field even with the JSON output, inconsistent exit codes, and the like.

The bottom line is, if somebody writes a checker for `Infer` I'll look at it, but I probably won't merge it in syntastic.  It's highly unlikely to solve all the problems mentioned above in a reasonable way, and IMO it's unlikely to be very useful to begin with.  Sorry about that.

**Comment by thyrgle** [comment]
No problem. Thank you for your time.
